### PR TITLE
Help 146 memtier benchmark

### PR DESCRIPTION
**Pull request by gordonca**
Added the document on Redis performance benchmarking to address the questions customers asked on [intercom](https://app.intercom.com/a/apps/reo7593q/inbox/inbox/conversation/67255500080004).

### CODE REVIEW

**Comment by laysauchoa** [comment]
Thanks for your contribution. Would not be easy if the user can follow the example and be able to reproduce the steps? We have similar tutorials in others HowTo that you can understand the format better, for example: https://docs.aiven.io/docs/products/mysql/howto/disable-foreign-key-checks.html

**Comment by gordonca** [comment]
@laysauchoa thanks for the comments. Please check the updated doc.

**Comment by laysauchoa** [comment]
Hi, it is getting better. Would be possible to make subsections like: 

- Prerequisites: What do the user needs here? I believe an Aiven Redis service running and a tool to be installed. 
- Running the benchmarking or commands

Also, can you go over the warnings? I believe there are some misspelling. Also, if you refer to code variables, can you make sure to add in between quotes like ``variable``.

About the code, you do not need to include the $ as the user may copy it and won't work when they run. Also can you split the command from the output? For the same previous reason, so it is easier to copy and run.  

You can find more instructions on the contributing guidelines here: https://github.com/aiven/devportal/blob/main/CONTRIBUTING.rst

**Comment by gordonca** [comment]
Hi Laysa,

Please check the new update committed.

I'm using vscode to edit the file, it does not give me warnings on
spelling. Where do you see the warning?

The '$' in the code is to indicate the line is a command, the following
lines are output. I separated the command and the output into two code
blocks, so the '$' can be removed.

Any more comments, please let me know.

Thanks,
Gordon

**Comment by laysauchoa** [comment]
Hi @gordonca you can see the warnings and errors in the checks by checking  this page: https://github.com/aiven/devportal/pull/1382/files You can see that a windows pops up with the error or warning about it. I will add one of the examples here, but you can find all the others in the page. 

Screenshot 2022-09-15 at 21 48 54

File: docs/products/redis/howto/benchmark-performance.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Redis®* **memtier_benchmark** is a command line utility developed by Redis Labs (formerly Garantia Data Ltd.) for load generation and bechmarking NoSQL key-value databases. It is recommended for the performance benchmarking of Aiven Redis®* services.
+
+.. Warning::
+    ``redis-benchmark`` is not supported to work with Aiven services, since ``CONFIG`` command is not allowed to run.
+    
+Prerequisites
+-------------
+* An Aiven Redis service running.
+* **memtier_benchmark** installed. To install the tool, first download the source code from `github <https://github.com/RedisLabs/memtier_benchmark>`_, and follow the instructions in ``README`` to install all the dependencies, then build and install the tool. Now it's ready to go.
```
Comment (laysauchoa) [inline docs/products/redis/howto/benchmark-performance.rst:12]:
```suggestion
* ``memtier_benchmark`` installed. To install the tool, first download the source code from `GitHub <https://github.com/RedisLabs/memtier_benchmark>`_, and follow the instructions in ``README`` to install all the dependencies, then build and install the tool. Now it's ready to go.
```

File: docs/products/redis/howto/benchmark-performance.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+.. Note::
+    The ``Testing`` section in ``README`` is not mandatory, can be skipped.
+
+Running benchmark
+-----------------
+Before using **memtier_benchmark**, read the help from :code:`mentier_benchmark -h` or this `Redis article <https://redis.com/blog/memtier_benchmark-a-high-throughput-benchmarking-tool-for-redis-memcached/>`_ to understand what the tool can offer.
```
Comment (laysauchoa) [inline docs/products/redis/howto/benchmark-performance.rst:19]:
```suggestion
Before using ``memtier_benchmark``, read the help from :code:`mentier_benchmark -h` or this `Redis article <https://redis.com/blog/memtier_benchmark-a-high-throughput-benchmarking-tool-for-redis-memcached/>`_ to understand what the tool can offer.
```

**Comment by gordonca** [comment]
Hi Laysa,

I updated again per your suggestion and the warnings.

Thank you for your patience.

-Gordon

**Comment by gordonca** [comment]
Hi @laysauchoa, I updated the doc 4 days ago, do you have more comments? Thanks.

**Comment by laysauchoa** [comment]
Hi, apologies for the delay. I will have time next week to review it again and test it.

**Comment by gordonca** [comment]
No worries. Thanks for the update.

**Comment by gordonca** [comment]
Hi Laysa,

I added the link to readme, also did some rewording to explain the sample
command's output roughly. It would be too verbose to explain the output in
detail. I expect readers can understand the meaning of the data if they
read the output carefully, since it's kind of self-explanatory.

Please let me know if you have more comments.

Thanks.

File: docs/products/redis/howto/benchmark-performance.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
+Benchmark performance
+=====================
+
+Redis®* **memtier_benchmark** is a command line utility developed by Redis Labs (formerly Garantia Data Ltd.) for load generation and bechmarking NoSQL key-value databases. It is recommended for the performance benchmarking of Aiven for Redis®* services.
```
Comment (laysauchoa) [inline docs/products/redis/howto/benchmark-performance.rst:4]:
```suggestion
Redis®* ``memtier_benchmark*`` is a command line utility developed by Redis Labs for load generation and performing benchmark NoSQL key-value databases. It is recommended for performing benchmark of Aiven for Redis®* services.
```

File: docs/products/redis/howto/benchmark-performance.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+.. Warning::
+    ``redis-benchmark`` is not supported to work with Aiven services, since ``CONFIG`` command is not allowed to run.
+    
+Prerequisites
+-------------
+* An Aiven for Redis service running.
+* ``memtier_benchmark`` installed. To install the tool, first download the source code from `Github <https://github.com/RedisLabs/memtier_benchmark>`_, and follow the instructions in the `README <https://github.com/RedisLabs/memtier_benchmark/blob/master/README.md>`_ to install all the dependencies, then build and install the tool. Now it's ready to go.
```
Comment (laysauchoa) [inline docs/products/redis/howto/benchmark-performance.rst:12]:
```suggestion
* ``memtier_benchmark`` installed. To install the tool, first download the source code from `GitHub <https://github.com/RedisLabs/memtier_benchmark>`_, and follow the instructions in the `README <https://github.com/RedisLabs/memtier_benchmark/blob/master/README.md>`_ to install all the dependencies, then build and install the tool. Now it's ready to go.
```

**Comment by laysauchoa** [comment]
Thank you for the fixes. We need to fix some warnings and some words before. I will check once again when we fix the warnings.

**Comment by gordonca** [comment]
Thank you for your contribution @laysauchoa 😄

**Pull request merged**